qml/SortTypeSelectionPage.qml

```diff
@@ -0,0 +1,69 @@
+import QtQuick 2.0
+import Sailfish.Silica 1.0
+//import Sailfish.Office 1.0
+import Sailfish.Office.Files 1.0
+
+Page {
+    id: root
+
+    signal sortSelected(int sortType)
+
+    function sortTypeText(sortType) {
+        if (sortType === FilterModel.Name) {
+            //: Sort by name
+            //% "Name"
+            return qsTrId("jolla-email-me-sort_name")
+        } else if (sortType === FilterModel.Type) {
+            //: Sort by type
+            //% "Type"
+            return qsTrId("jolla-email-me-sort_type")
+        } else if (sortType === FilterModel.Date) {
+            //: Sort by date
+            //% "Date"
+            return qsTrId("jolla-email-me-sort_date")
+        }
+
+        console.warn("Unknown sort type")
+        return ""
+    }
+
+    SilicaListView {
+        anchors.fill: parent
+        model: sortModel
+
+        header: PageHeader {
+            //% "Sort by:"
+            title: qsTrId("jolla-email-he-sort_by")
+        }
+
+        delegate: BackgroundItem {
+            Label {
+                x: Theme.horizontalPageMargin
+                anchors.verticalCenter: parent.verticalCenter
+                text: sortTypeText(sortType)
+                font.pixelSize: Theme.fontSizeMedium
+                color: highlighted ? Theme.highlightColor : Theme.primaryColor
+                elide: Text.ElideRight
+            }
+
+            onClicked: root.sortSelected(sortType)
+        }
+        VerticalScrollDecorator {}
+    }
+
+    ListModel {
```

Comment:
Small detail, but as this doesn't need any modifications to the list, could just do  model: [FilterModel.Name,  FilterModel.Type,  FilterModel.Date] and refer to model as modelData on the delegate.

Reply:
Done

Reply:
Done

Reply:
Done